Tests/test_peep_pip.py

```diff
@@ -0,0 +1,255 @@
+import unittest
+import math
+
+import numpy as np
+from scipy.interpolate import interp1d
+
+from peep_pip import PEEP, PIP
+from filter_rms_error import filter_rms_error
+from process_sample_data import ProcessSampleData
+
+
+fake_data_safety_factor = 8
+actual_data_safety_factor = 2
+
+
+class TestPEEP(unittest.TestCase):
+
+    def test_sin_input(self):
+        class PEEPTester(PEEP):
+            def __init__(self, dt):
+                super().__init__(dt, maximum_pressure=-0.9)
+
+        def desired_filter_data(t):
+            return np.full_like(t, -1)
+        normalized_error = filter_rms_error(PEEPTester,
+                                            np.sin,
+                                            desired_filter_data,
+                                            skip_initial=7,
+                                            use_pressure_error=True)
+        self.assertLess(normalized_error, 1/fake_data_safety_factor,
+                        "Fails to correctly calculate the PEEP of a "
+                        "sine.")
+
+    def test_cos_input(self):
+        class PEEPTester(PEEP):
+            def __init__(self, dt):
+                super().__init__(dt, maximum_pressure=-0.9)
+
+        def desired_filter_data(t): return np.full_like(t, -1)
+        normalized_error = filter_rms_error(PEEPTester,
+                                            np.cos,
+                                            desired_filter_data,
+                                            skip_initial=7,
+                                            use_pressure_error=True)
+        self.assertLess(normalized_error, 1/fake_data_safety_factor,
+                        "Fails to correctly calculate the PEEP of a "
+                        "cosine.")
+
+    def test_sin_greater_than_zero(self):
+        class PEEPTester(PEEP):
+            def __init__(self, dt):
+                super().__init__(dt, maximum_pressure=0.1)
+
+        def to_filter_data(t):
+            return np.maximum(np.sin(t), 0)
+
+        def desired_filter_data(t): return np.zeros_like(t)
+        normalized_error = filter_rms_error(PEEPTester,
+                                            to_filter_data,
+                                            desired_filter_data,
+                                            skip_initial=7,
+                                            use_pressure_error=True)
+        self.assertLess(normalized_error, 1/fake_data_safety_factor,
+                        "Fails to correctly calculate the PEEP of a "
+                        "sine where the range is clipped to the reals "
+                        "greater than 0.")
+
+    def test_approximate_breathing_data(self):
+        def to_filter_data(t):
+            # every 4th peak of max(20*sin(3t), 0)+4, plot:
+            # https://cdn.discordapp.com/attachments/610302955521966100/755223255836524624/unknown.png
+            return np.where((t % (3*20*np.pi/3/10)) < (20*np.pi/3/10),
```

Comment:
Where did this formula come from?

Reply:
I made it up.  I intended it to capture a realistic breathing waveform while still being "nice" data.